lib/src/kargs.rs

```diff
@@ -34,24 +35,27 @@ impl Config {
 /// Load and parse all bootc kargs.d files in the specified root, returning
 /// a combined list.
 pub(crate) fn get_kargs_in_root(d: &Dir, sys_arch: &str) -> Result<Vec<String>> {
+    // TODO optimize https://github.com/bytecodealliance/cap-std/pull/361
+    let d = &DirUtf8::from_cap_std(d.try_clone()?);
     // If the directory doesn't exist, that's OK.
     let Some(d) = d.open_dir_optional("usr/lib/bootc/kargs.d")? else {
         return Ok(Default::default());
     };
     let mut ret = Vec::new();
     // Read all the entries
-    let mut entries = d.entries()?.collect::<std::io::Result<Vec<_>>>()?;
-    // cc https://github.com/rust-lang/rust/issues/85573 re the allocation-per-comparison here
-    entries.sort_by_key(|a| a.file_name());
-    for ent in entries {
-        let name = ent.file_name();
-        let name = name
-            .to_str()
-            .ok_or_else(|| anyhow::anyhow!("Invalid non-UTF8 filename: {name:?}"))?;
-        if !Config::filename_matches(name) {
-            continue;
-        }
-        let buf = d.read_to_string(name)?;
+    let mut entries = d
+        .entries()?
+        .try_fold(Vec::new(), |mut acc, ent| -> Result<_> {
```

Comment:
This one can be replaced by `filenames_filtered` from https://github.com/coreos/cap-std-ext/pull/53